test/mock/cf-worker.mjs

```diff
@@ -0,0 +1,44 @@
+// mock cloudflare worker API:
+import { EventEmitter, once } from 'events'
+import API from '../../api/server.mjs'
+
+const events = new EventEmitter()
+
+// ENV
+globalThis.GITHUB_SECRET = 'fake-test-secret'
+
+// KV
+const NAN = (globalThis.NAN = { entries: {} })
+NAN.getWithMetadata = async (key) => NAN.entries[key]
+NAN.put = async (key, value, op) => (NAN.entries[key] = { ...op, value, key })
+NAN.list = async ({ prefix, limit = 1000 }) =>
+  Object.entries(NAN.entries)
+    .filter(([k]) => k.startsWith(prefix))
+    .slice(0, limit)
+    .map(([, v]) => v)
+
+globalThis.fetch = (url, request) =>
+  new Promise((resolve, reject) => {
+    events.once('error', reject)
+    events.emit('request', { url, request, reject, resolve })
+  })
+
+export const next = async () => (await once(events, 'request'))[0]
```

Comment:
this `next` function is a way to capture the next `fetch` request made by the worker